app/controllers/locator_maps_controller.rb

```diff
@@ -0,0 +1,35 @@
+class LocatorMapsController < ApplicationController
+  def index
+    # center the map by default on the rough middle of CONUS
+    @origin = [38.0, -97.0]
+    @offices = []
+    @offset = params[:offset].present? ? params[:offset].to_i : 0
+    if @offset < 0
+      @offset = 0
+    end
+
+    if params[:zipcode].present?
+      if params[:zipcode].match('\d{5}') != nil
+        search_zipcode
+      end
```

Comment:
These lines could possibly be expressed as:

```rb
search_zipcode if params[:zipcode].match(/\d{5}/).length
```

Reply:
The nil check is still necessary, but otherwise, I agree.